ls1intum/Artemis
# `Quiz exercises`: Require a confirmation when ending quiz exercises

### Checklist
#### General


#### Server
#### Client
guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).

### Motivation and Context
Followup for #4936.
Ending a quiz cannot be reversed and didn't even use a danger button.

### Description
The end now button now is no longer available to editors, is now danger red and requires a confirmation when pressed.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Editor
- 1 Tutor

1. Create a quiz exercise in Batched or Individual mode
2. Set the quiz to be visible to students
3. Check that instructors can see the End Now button in the Quiz Lifecycle column.
4. Check that the button appears to be sufficiently dangerous to press.
5. Check that Editors and Tutors cannot access the button.
6. Check that when pressing the button as an instructor a confirmation dialog appears that sufficiently explains the danger of pressing the button.
7. Check that after confirming the dialog the quiz is now in the closed state
8. Delete the quiz

NOTE: If you test the fact that this interrupts running quiz batches, the students will get no indication that the quiz was just ended other than that all further autosaves and the submit will fail.

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


| Class/File | Branch | Line |
|------------|-------:|-----:|


### Screenshots


![image](https://user-images.githubusercontent.com/44437514/168110597-f68eded6-5eaa-4cc6-a827-507989337287.png)
![image](https://user-images.githubusercontent.com/44437514/168110664-91b51a83-e100-4e64-9d5e-f8ef9ce39dc6.png)
